CONTRIBUTING.md

```diff
@@ -1,106 +1,186 @@
- 6. **MANDATORY**: Add or update tests for the newly generated code.
- 7. Once added to the Azure SDK for .NET, build your local package using command
- e.g.
- `msbuild build.proj /t:CreateNugetPackage /p:scope=Compute`
- 8.  A Pull request of your Azure SDK for .NET changes against **master** branch of the [Azure SDK for .NET](https://github.com/azure/azure-sdk-for-net)
- 9. Both the pull requests will be reviewed and merged by the Azure SDK team
+1.  Create fork of [Azure REST API Specs](https://github.com/azure/azure-rest-api-specs)
+2.  Create fork of [Azure SDK for .NET](https://github.com/azure/azure-sdk-for-net)
+3.  Create your Swagger specification for your HTTP API. For more information see
+    [Introduction to Swagger - The World's Most Popular Framework for APIs](http://swagger.io)
+4.  Install the latest version of AutoRest and use it to generate your C# client. For more info on getting started with AutoRest,
+    see the [AutoRest repository](https://github.com/Azure/autorest)
+5.  Create a branch in your fork of Azure SDK for .NET and add your newly generated code to your project. If you don't have a project in the SDK yet, look at some of the existing projects and build one like the others.
+6.  **MANDATORY**: Add or update tests for the newly generated code.
+7.  Once added to the Azure SDK for .NET, build your local package using [client](#client-libraries) or [management](#management-libraries) library instructions shown in the above sections.
+8.  A Pull request of your Azure SDK for .NET changes against **master** branch of the [Azure SDK for .NET](https://github.com/azure/azure-sdk-for-net)
+9.  The pull requests will be reviewed and merged by the Azure SDK team
 
 ### New Resource Provider
+
 1. If you have never created an SDK for your service before, you will need the following things to get your SDK in the repo
 2. Follow the standard process described above.
-3. Directory names helps in using basic heuristics in finding projects as well it's associated test projects during CI process.
-4. Create a new directory (name of your service e.g. Compute, Storage etc)
-6. For e.g. for management plane SDKs the following directory structure
- - `<RPName>\<packageName>\src\Microsoft.Azure.Management.<RPName>\Microsoft.Azure.Management.<RPName>.csproj`
- - `<RPName>\<packageName>\tests\Microsoft.Azure.Management.<RPName>.Tests.csproj`
+3. Project names helps in using basic heuristics in finding projects as well it's associated test projects during CI process.
+4. Create a new directory using the name of your service/ resource provider as specified in [azure-rest-api-specs/specification](https://github.com/Azure/azure-rest-api-specs/tree/master/specification) Repo
+5. Follow the the directory structure below
+
+```
+sdk\<service name>\<package name>\README.md
+sdk\<service name>\<package name>\*src*
+sdk\<resource provider name>\<package name>\*tests*
```

Comment:
This should be `<service name>` in all the samples we should have some be service and others be resource provider.

Reply:
Do you mean shouldn't? i.e. 'shouldn't have some be service and others be resource'

Reply:
Yes edited :)